documentation/managing_modules.md

```diff
@@ -0,0 +1,183 @@
+# 🧪 Managing modules in Bolt projects
+
+Bolt version 2.30.0 introduced module workflows for Bolt projects that improve
+dependency management and make it easier to add new modules to your Bolt
+project. These workflows, which include the `bolt module install`, `bolt module
+add`, and `bolt project migrate` commands, are experimental.
+
+## What's changed?
+
+- **Module installation directory:** Bolt now installs modules into the
+  `.modules/` directory instead of `modules/`. Avoid committing `.modules/` to
+  source control. Your users can download your Bolt project and use the `bolt
+  module install` command to download the required modules from the forge.
+- **Module configuration:** Your `bolt-project.yaml` file contains a `modules`
+  key that lists the direct module dependencies of your Bolt project. Unless you
+  need to pin a module to a specific version, avoid editing the `modules` list
+  directly. Instead, use the `bolt module add` command.
+- **Puppetfile:** The Puppet file is now a lock file. Bolt generates a
+  Puppetfile each time you modify your modules with a Bolt command. Do not edit
+  the Puppetfile directly. Instead, use Bolt commands to manage your modules,
+  and rely on Bolt to manage the Puppetfile. You can compare a Puppetfile to a
+  `Gemfile.lock` file in Ruby, or a `yarn.lock` file in Yarn.
+- **Module path:** The new module path is `['modules']` and Bolt always appends
```

Comment:
Looks like Puppet uses `modulepath` https://puppet.com/docs/puppet/6.18/dirs_modulepath.html we seem to use `module path` and `modulepath` pretty much interchangeably in the Bolt docs. I'm fine with either, it'd be good to standardize.

Reply:
Yeah, good point. There are a couple of other things we should standardize too. I'll add this to the list and open a ticket for those.

Reply:
https://github.com/puppetlabs/bolt/pull/2225 Should do it.